### PR TITLE
Feature/merge dtmf dead actor fix

### PR DESCRIPTION
**Pull request by runningferret**
Cherry pick ggayans process_dtmf fix to not blow up on a dead actor exception when dealing with dtmf destined for a dead actor

### CODE REVIEW

**Comment by runningferret** [comment]
@sfgeorge @jared-prime

**Comment by Jared-Prime** [comment]
cool :+1:

**Comment by Jared-Prime** [comment]
@runningferret I'm still OK with this PR, but in your opinion, are we certain this will fix the specific error we've encountered? ( I know you mentioned this during scrum, but I sometimes forgets :smile_cat: )

**Comment by runningferret** [comment]
No, I do not believe this will fix the specific process_dtmf issue we have seen in PROD since releasing Phoenix. In theory this is still something we could run into, but what it looks like at first blush is we are using a ComposedPrompt, whereas this patch was for someone using just a plain Input. They're both subclasses of Component, so they behave very similarly, but one difference is that ComposedPrompt provides its own implementation of process_dtmf and then calls super (into InputComponent). We see the DeadActor in the ComposedPrompt portion, before it calls super and gets into InputComponent (where this fix is applied). 

tl;dr: I think we can still get hit by the issue this patch addresses, but we hit a DeadActor error slightly earlier in ComposedPrompt and throw an exception there. I still think there is value in bringing this into our fork, as well as investigating if we should craft a similar patch around ComposedPrompt#process_dtmf

**Comment by Jared-Prime** [comment]
> I still think there is value in bringing this into our fork

agreed!

**Comment by sfgeorge** [comment]
I'd prefer to only bring in the changes that we need to.  If we don't trigger this with our prompt usage, I think the cost of further divergence from the main Punchblock repo outweighs the benefit.

If, however, another similarly fashioned rescue actually reduced the DeadActor error rate for us, it would be worthwhile.

**Comment by runningferret** [comment]
Closing due to we don't want the risk since it doesn't explicitly fix any problems we have.
